lib/ansible/plugins/action/package.py

```diff
@@ -56,6 +56,14 @@ def run(self, tmp=None, task_vars=None):
                 module = facts.get('ansible_facts', {}).get('ansible_pkg_mgr', 'auto')
 
             if module != 'auto':
+                if module not in ['apk', 'apt_rpm', 'apt', 'dnf', 'homebrew_cask',
+                                  'homebrew_tap', 'homebrew', 'installp', 'macports', 'mas',
+                                  'openbsd_pkg', 'opkg', 'pacman', 'pkg5', 'pkgin',
+                                  'pkgng', 'pkgutil', 'portage', 'portinstall', 'slackpkg',
+                                  'snap', 'sorcery', 'svr4pkg', 'swdepot', 'swupd',
+                                  'urpmi', 'xbps', 'yum', 'zypper']:
```

Comment:
I'm not sure this is the correct location to do this check.

I think we need to do the checks in the `if module == 'auto':` statements, to make sure that `ansible_pkg_mgr` returned is valid. I think the concern is a compromised machine that returns a value that causes us to automatically use an invalid package module.

Although `use` could be abused by this, that seems like more of a explicit effort, as opposed to implicit behavior.